aip/general/0213.md

```diff
@@ -9,44 +9,89 @@ placement:
-into taking major version changes or even lead to dependency conflicts.
+As specified in [AIP-215][], APIs must be self-contained except for the use
+of "common component" packages which are intended for use by multiple APIs.
 
-However, there are also cases where common structures are valuable, especially
-where a concept is well-known and it is sufficiently clear that it will not
-change. Common protos serve this use case.
+There are two kinds of common component packages:
 
-## Guidance
+- Organization-specific common components, covering organization-specific
+  concepts such as a "viewport" in Maps.
+- Global common components which are generic (i.e. not domain-specific),
+  such as "timestamp" or "postal address".
 
-The public representation of APIs **should** be self-contained, meaning that
-all protos used by the API originate in the same proto `package`, except for
-common protos, which **may** be used freely in any API.
+Where it is safe to share a single representation across multiple APIs,
+common components can make it easier for clients to interact with those APIs.
+Concept-specific client code can be written once, and messages can be used from
+the response of one API in the request of another without clunky copying, for
+example.
 
-APIs **must not** define a set of API-specific common protos which live outside
-of its versioning structure. This prevents independent movement of particular
-versions and also causes problems for client libraries in many languages that
-compile the proto messages into classes.
+This benefit comes with significant restrictions and limitations, however,
+and should not be attempted lightly.
 
-APIs **should not** directly depend on protos defined in other APIs. Instead,
-they should copy and paste the applicable messages into their own API. When
-doing so, APIs **should** keep the field names and numbers the same.
+Note that even if the *domain* of a component is common, the requirements of
+a component may be organization-specific. For example, some organizations may
+have particular requirements of how financial values are represented, leading
+to multiple finance-oriented organization-specific common components - because
+any global common component would either not meet the organization-specific
+requirements, or be too complex for general use.
 
-## Existing common protos
+## Guidance
 
-The common protos, which public-facing protos for an API **may** safely import,
-are as follows:
+- Organization-wide common component packages **must** end with `.type`, e.g.
+  `google.geo.type` or `google.shopping.type`.
+- Organizations **must** consult the API design team before creating a new
+  organization-wide common component package.
+- Organizations **must not** define generic components in organization-specific
+  common component packages, instead preferring global common components.
+- Common components **must not** be "moved" from an organization-specific common
```

Comment:
I think it'd be nice to define "move" a bit more here - is completely copying the proto not allowed? if not, what happens when a concept does become common across multiple protos?

Reply:
Added clarification, and a "may" for copy-without-removal.